### DISCUSSION TITLE
Change folder name for exported images and labels

### DISCUSSION THREAD

**Discussion by YezCohen** [ultralytics/yolov5#10095] [Q&A]
Hi there,

I searched for an answer to this question in previous discussions and couldn't find anything helpful.

When running detect.py, with subprocess.run() I transfer all the arguments needed for my case (--weights, --img, --conf, --source, --save-txt) and it runs perfectly fine and exports all the annotated images and labels to the directory I chose in a folder named 'exp' (which is the default for the --name argument for detect.py.

Now, I want that every time I run detect.py I will pass the --name argument so that the name of the folder for exported files won't be 'exp' but the name I chose.

For some reason, it still saves all the files to a folder named 'exp'.

If anyone knows how to help me, tell me where I went wrong, or that I didn't understand the purpose of the --name argument - I would appreciate any kind of help.

Thank you!

**Update:** When trying to debug it, I noticed that even though I'm sending in subprocess.run() a list of all the arguments including the --name argument:
`subprocess.run(
        ['python3', './yolov5/detect.py', '--weights', 'path/to/weights', '--img', '416', '--conf',
        '0.4', '--source', 'path/to/source', '--name', 'FOLDER_NAME', '--save-txt'])`

the sys.argv list should be in length 12 (including --name and FOLDER_NAME) 
but when I print the args list in the detect.py script I get only 10 arguments when it's ignoring the --name and FOLDER_NAME like it didn't receive them.

All help would be greatly appreciated!

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://github.com/ultralytics/yolov5/issues/251) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details.

Good luck 🍀 and let us know if you have any other questions!

**Reply by YezCohen** [reply]
> 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models allow for simple model loading and inference in a pure python environment without using `detect.py`.
> 
> ### Simple Inference Example
> This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://github.com/ultralytics/yolov5/issues/251) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.
> 
> ```python
> import torch
> 
> # Model
> model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
> #                                            'custom', 'path/to/best.pt')  # custom model
> 
> # Images
> im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list
> 
> # Inference
> results = model(im)
> 
> # Results
> results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
> results.xyxy[0]  # im predictions (tensor)
> 
> results.pandas().xyxy[0]  # im predictions (pandas)
> #      xmin    ymin    xmax   ymax  confidence  class    name
> # 0  749.50   43.50  1148.0  704.5    0.874023      0  person
> # 2  114.75  195.75  1095.0  708.0    0.624512      0  person
> # 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
> 
> results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
> # person    2
> # tie       1
> ```
> 
> <img alt="" width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">
> 
> See [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details.
> 
> Good luck 🍀 and let us know if you have any other questions!

Thank you for the detailed response.
I do not want to use the model in pure python environment.

The only thing I want to do is change / add to the source code that the folder for the exported annotated images won't be named 'exp' but a different name that I choose. I assume it is something that is possible to get done since there is a parameter named '--name' to run the model.

Is it possible to do this?
Thank you!

**Reply by glenn-jocher** [reply]
@YezCohen yes for that you can use `python detect.py --project PROJECT --name NAME`, which will save results to PROJECT/NAME

**Reply by YezCohen** [reply]
Thank you for the response! 
So I tried only adding --name NAME and it did not work, is there a chance that I need both --project and  --name?
Because when I add only --name it ignores it.

I'll try it.

**Reply by glenn-jocher** [reply]
@YezCohen nothing is ignored. If you pass --name it is used as your experiment name.
